cli/cli/Commands/Project/GenerateClientFileCommand.cs

```diff
@@ -35,48 +35,73 @@ public override void Configure()
 	{
 		#region load client dll into current domain
 
-		var absolutePath = Path.GetFullPath(args.microserviceAssemblyPath);
-		var absoluteDir = Path.GetDirectoryName(absolutePath);
-		var loadContext = new AssemblyLoadContext("generate-client-context", false);
-		loadContext.Resolving += (context, name) =>
+		// Get the list of all existing microservices
+		var allServices = args.BeamoLocalSystem.BeamoManifest.ServiceDefinitions.Where(sd => sd.Protocol is BeamoProtocolType.HttpMicroservice).ToArray();
+
+		// Get the list of dependencies of each of these microservices
+		var allDepTasks = allServices.Select(sd => args.BeamoLocalSystem.GetDependencies(sd.BeamoId)).ToArray();
+		var allDeps = await Task.WhenAll(allDepTasks);
+
+		// Get the list of all BeamoIds whose DLLs we need to have loaded for a single pass.
+		var allServicesToLoadDlls = allServices.Select(sd => sd.BeamoId).Union(allDeps.SelectMany(d => d)).Distinct().ToArray();
+
+		// Get the list of all assemblies paired with their last edit time.
+		var allAssemblies = new List<Assembly>();
+		foreach (string beamoId in allServicesToLoadDlls)
 		{
-			var assemblyPath = Path.Combine(absoluteDir, $"{name.Name}.dll");
-			try
+			var isProjBuilt = await ProjectCommand.IsProjectBuilt(args, beamoId);
+			if (isProjBuilt.isBuilt)
 			{
-				Log.Verbose($"loading dll name=[{name.Name}] version=[{name.Version}]");
-				if (assemblyPath != null)
-					return context.LoadFromAssemblyPath(assemblyPath);
-				return null;
-			}
-			catch (Exception ex)
-			{
-				BeamableLogger.LogError($@"Unable to load dll at path=[{assemblyPath}] 
+				var dllPath = isProjBuilt.path;
+				var absolutePath = Path.GetFullPath(dllPath);
+				var absoluteDir = Path.GetDirectoryName(absolutePath)!;
+				var loadContext = new AssemblyLoadContext($"generate-client-context-{beamoId}", false);
+				loadContext.Resolving += (context, name) =>
+				{
+					var assemblyPath = Path.Combine(absoluteDir, $"{name.Name}.dll");
+					try
+					{
+						Log.Verbose("loading dll name=[{Name}] version=[{Version}]", name.Name, name.Version);
+						var loadedDependentAsm = context.LoadFromAssemblyPath(assemblyPath);
+						allAssemblies.Add(loadedDependentAsm);
+						return loadedDependentAsm;
+					}
+					catch (Exception ex)
+					{
+						BeamableLogger.LogError($@"Unable to load dll at path=[{assemblyPath}] 
 name=[{name}] 
 context=[{context.Name}]
 message=[{ex.Message}]
 ex-type=[{ex.GetType().Name}]
 inner-message=[{ex.InnerException?.Message}]
 inner-type=[{ex.InnerException?.GetType().Name}]
 ");
-				throw;
+						throw;
+					}
+				};
+
+				var userAssembly = loadContext.LoadFromAssemblyPath(absolutePath);
+				Log.Verbose("loading dll name=[{Name}] version=[{Version}] deps=[{Deps}]", userAssembly.GetName().Name, userAssembly.GetName().Version, string.Join(", ",userAssembly.GetReferencedAssemblies().Select(n => n.Name)));
+				
+				/// GHOST IN THE MACHINE ---> We need some time to investigate this stuff.

```

Comment:
:( 👻

Reply:
Yeah --- this assembly loading thing has got me a bit worried. This is working correctly though... it just worries me that it should be working without that line as far as I can tell.